src/humanPose/HumanPoseRenderInstance.js

```diff
@@ -208,10 +211,18 @@ export class HumanPoseRenderInstance {
         }
         this.pose.forEachJoint(joint => {
             this.lensColors[lens.name].joints[JOINT_TO_INDEX[joint.name]] = lens.getColorForJoint(joint);
+            if (!joint.valid) {
+                this.lensColors[lens.name].joints[JOINT_TO_INDEX[joint.name]] = MotionStudyColors.undefined;
+            }
         });
         this.pose.forEachBone(bone => {
             this.lensColors[lens.name].bones[BONE_TO_INDEX[bone.name]] = lens.getColorForBone(bone);
+            if (!bone.valid) {
+                this.lensColors[lens.name].bones[BONE_TO_INDEX[bone.name]] = MotionStudyColors.undefined;
+            }
         });
+        // MK - why this condition (lens === this.lens)? When switching lens this is not true and this is not applied. 
```

Comment:
It's been a while, so I'm not 100% sure, but I think this was meant for live pose data so that when a new pose is added it only updates the selected lens joint colors for efficiency?